### PR TITLE
refactor: [TECH-1483] dataElements store in IndexedDB

### PR DESCRIPTION
**Pull request by JoakimSM**
Extract data elements into its own IndexedDB store.

A major reason for doing this right now is that we will be needing the data elements metadata in the relationships (and we will have to get them by id).

### CODE REVIEW

**Comment by JoakimSM** [comment]
(I will fix the conflict when it is time to merge, for every new release the conflict will reappear)

File: cypress/integration/EnrollmentEditEventPage/EnrollmentEditEventPageForm.feature [hunk 1]
```diff
@@ -73,14 +73,14 @@ And the user see the following text: 11
...
 
-Scenario: User can update schedule date for a scheduled event 
-    Given you land on the enrollment event page with selected person by having typed /#/enrollmentEventEdit?eventId=RIrfCcEP8Uu&orgUnitId=DiszpKrYNg8
-    Then the user see the following text: Enrollment: Edit Event
-    And the user see the following text: Infant Feeding
-    When the user clicks switch tab to Schedule
-    And the user selects another schedule date
-    And the user clicks on the schedule button on widget-enrollment-event
-    Then the user see the following text: Enrollment Dashboard
+# Scenario: User can update schedule date for a scheduled event
```
Comment (simonadomnisoru) [inline cypress/integration/EnrollmentEditEventPage/EnrollmentEditEventPageForm.feature:76]: Hey, any reason why this test was disabled?

Comment (JoakimSM) [reply cypress/integration/EnrollmentEditEventPage/EnrollmentEditEventPageForm.feature:76]: Not any more, so good catch 👍
We had this problem here 3 weeks ago: https://dhis2.atlassian.net/browse/DHIS2-14301. Will readd test.

**Review by simonadomnisoru — APPROVED** [review]

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.40.0 version

**Pull request merged**